src/libraries/System.IO.Compression/src/System/IO/Compression/ZipArchive.cs

```diff
@@ -211,6 +214,18 @@ public ZipArchive(Stream stream, ZipArchiveMode mode, bool leaveOpen, Encoding?
             }
         }
 
+        /// <summary>
+        /// Gets or sets the optional archive comment.
+        /// The comment encoding is determined by the <c>entryNameEncoding</c> parameter of the <see cref="ZipArchive(Stream,ZipArchiveMode,bool,Encoding?)"/> constructor.
+        /// </summary>
+        [AllowNull]
+        public string Comment
+        {
+            get => _archiveCommentAsString ?? string.Empty;
```

Comment:
`_archiveCommentAsString` exists only to make it faster to get Comment repeatedly, right? Is that something we expect people to do? It might be better to just decode into a string when requested, and you make the code a lot simpler.

Reply:
Logic changed, based on this suggestion and to Adam's.